### PR TITLE
debug, and fix problem

### PR DESCRIPTION
**Pull request by vladislavprovich**
fix problem, and test request

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -14,46 +16,51 @@ import (
...
 
 func main() {
 	// Initialize
-	logger := initLoger()
+	logger := initLogger()
 	db := initDB()
 	defer func() {
 
 		if err := db.Close(); err != nil {
-			logger.Warn("Error closing db", zap.Error(err))
```
Comment (vkotsiuba99) [inline cmd/server/main.go:29]: why remove Warn

File: cmd/server/main.go [hunk 2]
```diff
@@ -14,46 +16,51 @@ import (
...
-
 	repo := initRepo(db)
 	service := initService(&repo, logger)
 	urlHandler := initHandler(service, logger)
 
+	logger.Debug("main, db initialized", zap.Any("db", db), zap.Any("repo", repo))
+	logger.Debug("main, logger initialized", zap.Any("logger", logger))
+	logger.Debug("main, repo initialized", zap.Any("repo", repo))
+	logger.Debug("main, service initialized", zap.Any("service", service))
+	logger.Debug("main, URL handler initialized", zap.Any("URL_handler", urlHandler))
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: remove all debugs

File: cmd/server/main.go [hunk 3]
```diff
@@ -14,46 +16,51 @@ import (
...
+
 	// Create Router
 	r := chi.NewRouter()
-
+	r.Use(chiMiddleware.Logger)
 	// Apply Middlewares
-	r.Use(middleware.Recoverer(logger))
+	r.Use(middleware.Recoverer(logger)) // is working
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: remove comments

File: cmd/server/main.go [hunk 4]
```diff
@@ -14,46 +16,51 @@ import (
...
+	logger.Debug("main, db initialized", zap.Any("db", db), zap.Any("repo", repo))
+	logger.Debug("main, logger initialized", zap.Any("logger", logger))
+	logger.Debug("main, repo initialized", zap.Any("repo", repo))
+	logger.Debug("main, service initialized", zap.Any("service", service))
+	logger.Debug("main, URL handler initialized", zap.Any("URL_handler", urlHandler))
+
 	// Create Router
 	r := chi.NewRouter()
-
+	r.Use(chiMiddleware.Logger)
```
Comment (vkotsiuba99) [inline cmd/server/main.go:37]: must be with all Middlewares

File: cmd/server/main.go [hunk 5]
```diff
@@ -14,46 +16,51 @@ import (
...
+	r.Use(middleware.Recoverer(logger)) // is working
 	r.Use(middleware.RequestLogger(logger))
 	r.Use(middleware.CORS)
 	r.Use(middleware.RateLimiter)
 
 	// Routes
-	r.Post("/shorten", urlHandler.ShortenURL)
-	r.Get("/{shortURL}", urlHandler.Redirect)
-	r.Get("/{shortURL}/stats", urlHandler.GetStats)
+	r.Post("/shorten", urlHandler.ShortenURL)       // is working
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: delete comments like this everywhere

File: cmd/server/main.go [hunk 6]
```diff
@@ -14,46 +16,51 @@ import (
...
+	r.Get("/{shortURL}", urlHandler.Redirect)       // is working
+	r.Get("/{shortURL}/stats", urlHandler.GetStats) // is working
 
 	// Start Server Graceful Shutdown
 	serverPort := os.Getenv("SERVER_PORT")
 	if serverPort == "" {
 		serverPort = "8080"
+		logger.Debug("setting server port", zap.String("port", serverPort))
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: remove debug everywhere

File: cmd/server/main.go [hunk 7]
```diff
@@ -105,9 +111,8 @@ func initRepo(db *sql.DB) repository.URLRepository {
 
 func initService(repo *repository.URLRepository, logger *zap.Logger) service.URLService {
 	return service.NewURLService(*repo, logger)
-
 }
 
-func initHandler(srv service.URLService, logger *zap.Logger) *handler.URLHandler {
-	return handler.NewURLHandler(srv, logger)
+func initHandler(serv service.URLService, logger *zap.Logger) *handler.URLHandler {
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: srv, not serv
write shortly "srv" or full "service"

File: internal/handler/url_handler.go [hunk 8]
```diff
@@ -19,19 +19,20 @@ type URLHandler struct {
 	logger  *zap.Logger
 }
 
-func NewURLHandler(srv service.URLService, logger *zap.Logger) *URLHandler {
+func NewURLHandler(serv service.URLService, logger *zap.Logger) *URLHandler {
```
Comment (vkotsiuba99) [inline internal/handler/url_handler.go]: same 
srv, not serv
write shortly "srv" or full "service"

File: internal/handler/url_handler.go [hunk 9]
```diff
@@ -19,19 +19,20 @@ type URLHandler struct {
...
 	return &URLHandler{
-		service: srv,
+		service: serv,
```
Comment (vkotsiuba99) [inline internal/handler/url_handler.go]: fix

File: internal/handler/url_handler.go [hunk 10]
```diff
@@ -79,7 +87,8 @@ func (h *URLHandler) Redirect(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Log the redirect
+	// Log the redirect //todo fix this code
```
Comment (vkotsiuba99) [inline internal/handler/url_handler.go]: delete this comments

File: internal/repository/postgres.go [hunk 11]
```diff
@@ -1,8 +1,11 @@
 package repository
 
-import "database/sql"
+import (
+	"database/sql"
+)
 
 func InitDB(connStr string) (*sql.DB, error) {
+	//connStr = "user=postgres password=password dbname=urlshortener host=db sslmode=disable host=localhost"
```
Comment (vkotsiuba99) [inline internal/repository/postgres.go]: remove, env vars must be in .env file, not in the project code

File: cmd/server/main.go (hunk 1 above)
Comment (vladislavprovich) [reply cmd/server/main.go:29]: my bad

File: cmd/server/main.go (hunk 4 above)
Comment (vladislavprovich) [reply cmd/server/main.go:37]: ok

File: cmd/server/main.go (hunk 7 above)
Comment (vladislavprovich) [reply cmd/server/main.go]: yes, i am testing different options

File: internal/handler/url_handler.go (hunk 9 above)
Comment (vladislavprovich) [reply internal/handler/url_handler.go]: all ser fixed

Comment (vladislavprovich) [reply internal/handler/url_handler.go]: srv*

File: internal/repository/postgres.go (hunk 11 above)
Comment (vladislavprovich) [reply internal/repository/postgres.go]: i used it for local test

File: cmd/server/main.go (hunk 2 above)
Comment (vladislavprovich) [reply cmd/server/main.go]: sorry, i tnink needed remove only this debug

**Pull request merged**